test-adjuster/adjuster.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"strings"
+	"sync"
+	"time"
+)
+
+var (
+	repoPath      = flag.String("repo-path", "/kubernetes", "Path to the clone of Kubernetes repository.")
+	repoLock      = sync.Mutex{}
+	tokenFile     = flag.String("token-file", "/token", "Path to a file with saved github token.")
+	pollFrequency = 10 * time.Minute
+	checkers      = []*Checker{}
+)
+
+func readToken() string {
+	data, err := ioutil.ReadFile(*tokenFile)
+	if err != nil {
+		panic(err)
+	}
+	return strings.TrimSpace(string(data))
+}
+
+func registerChecker(trigger Trigger, action Action) {
+	checkers = append(checkers, NewChecker(trigger, action))
+}
+
+func lockAndGetRepo() string {
+	repoLock.Lock()
+	return *repoPath
+}
+
+func unlockRepo() {
+	repoLock.Unlock()
+}
+
```

Comment:
Can you put this into a class instead of random individual functions?
I would like to avoid things like global locks, and stuff like that.